### DISCUSSION TITLE
Call onCreate hook when inserting using EntityManager#insert

### DISCUSSION THREAD

**Discussion by Kalos-S** [mikro-orm/mikro-orm#4677] [Ideas]
**Is your feature request related to a problem? Please describe.**
When creating an entity using EntityManager.create and saving it to the database using EntityManager.insert the onCreate hooks of the entities properties are not called, potentially causing errors or invalid entities being inserted.

**Describe the solution you'd like**
Execute onCreate hooks when inserting entities or provide an alternate api that does. Alternatively add the option to have the default value be a function.

**Describe alternatives you've considered**
Using an IIFE seems like an options but is more of a hack.

**Comment by B4nan** [comment]
Hooks are handled by the UoW, you need to use `em.flush` to trigger them. `em.insert` is just a shortcut for query builder, and as such, it does not fire any hooks. Moreover, `em.insert` does not require passing entity instances, it accepts POJOs as well - and we need entity instances for the hooks, so having the hooks in there would mean having different behavior based on the input. I can't say I like that idea... 

Btw `onCreate` is not really a hook, hooks are the `before/afterCreate` etc... But this is all handled by the UoW, including the `onCreate/onUpdate` property options.

Not sure how IIFE is related to this?

---

Changing this behavior is imo breaking, I am not completely opposed to doing this, but it would have to go to v6. I'd also argue this should be handled in the `em.create` call instead of `em.insert` (where we might not be working with entity instances at all).

**Comment by Kalos-S** [comment]
Hey! I appreciate the quick reply.

I agree with you on those points. Moving the onCreate call to em.create seems to be the most intuitive way for handling this and would probably be the best solution for this.
